### ISSUE TITLE
Vis Network: Manipulation - click on "Add Edge" button event?

### ISSUE THREAD

**Issue by FilipQL** [almende/vis#3182]
I am using the `hoverNode` event to dynamically set the value of the node **title** (_tooltip_, _popup_), as discussed here: https://github.com/almende/vis/issues/3157

```
network.on("hoverNode", function (params) {
    if(document.getElementById('lab-popover-checkbox').checked) { // if popovers are enabled
        let node = data.nodes.get(params.node);
        network.canvas.body.nodes[params.node].options.title = DSL.generateTooltipNodeScript(node);
    } else {
        network.canvas.body.nodes[params.node].options.title = null;
    }
});
```
Since these titles (tooltips, popovers) can be very big - it can be really irritating when adding edges. So want the following:

When a user clicks on  the '_addEdge_' button in the GUI -> temporarily disable node titles. Finally, when adding edge is finished (edge is added) OR canceled (user clicked on the "_back_" button in the GUI) - I need to enable node titles again.

In the [documentation](http://visjs.org/docs/network/#Events) there are no such events, so is this possible in any way?

**Comment by wimrijnders** [comment]
I would propose temporarily setting the label delay to a large value while editing:

```
var options = {
  interaction: {
    tooltipDelay: 3600000    // 1 hour, YMMV
  }
}
```
... and then return it to the original value when done with editing.

**Comment by FilipQL** [comment]
@wimrijnders How to "track" if a user is adding a new edge (if user **clicked** on the '_Add Edge_' button in the menu)?

**Comment by wimrijnders** [comment]
I've been looking into this, and I can't figure it out.

Things I tried:

### 1. Adding a general click handler

```js
document.addEventListener('click', function (e) {
  var hasClass = function(elem, className) {
    return elem.className.split(' ').indexOf(className) > -1;
  }

  if (hasClass(e.target, 'vis-connect')) {
    alert('click');
  }
});
```
Here, `vis-connect` is the class for the 'Add Edge' button.
This handler is called for everything expect the manipulation buttons. I think that `hammer.js` intercepts all of the clicks for these buttons.

### 2. Use the `addEdge` callback

```js
var options = {
  manipulation: {
    enabled: true,
    addEdge: function(edgeData,callback) {
      alert("Called 'Add Edge'");
    }
  }
};
```

This fires at the moment that an edge has been dragged between nodes, right before creating that edge. This is not connected to clicking the "Add Edge" button.


### 3. Intercept the initialization of 'add Edge' mode.

This happens in `ManipulationSystem.addEdgeMode()`.
This fires at the correct moment; however, there is no event fired when it is run, and there is no legal way to intercept this call.

----

So, in conclusion, I really can't tell you how to track a new edge action. Sorry. Perhaps a wiser person than me can answer your question.

**Comment by wimrijnders** [comment]
@FilipQL Can this issue be closed now? Sorry to be of limited help.

**Comment by wimrijnders** [comment]
Thanks!
